### ISSUE TITLE
Cannot finish update to 3.2.4revised; R cannot find the location of the new R to update packages

### ISSUE THREAD

**Issue by bharris-fs** [talgalili/installr#53]
I installed a new copy of installr v. 0.17.5 today and ran it to update R from 3.2.3 to 3.2.4revised on W7.

I then opened RGui (64-bit) as admin, attached installr with library(), and ran updateR().  It told me there was a newer version: 3.2.4 (2016-03-17).  I accepted that, declined reading the news, and said yes to installing R.  It downloaded and ran R-3.2.4revised.exe.

I then set up English, clicked Next, and clicked Next.  Then I Browsed to C:\R\R-3.2.4revised\, clicked Next, (allowed to install over previous attempts,) left the default on selecting components (all), specified No (accept defaults), clicked Next for start menu folder, clicked next for defaults, and let it install.  It seemed to be installing in the C:\R\R-3.2.4revised\, as requested.

I then clicked Finish.

installr complained that it couldn't find the location of the new R.  Here is the entire trace.

`R version 3.2.3 (2015-12-10) -- "Wooden Christmas-Tree"
Copyright (C) 2015 The R Foundation for Statistical Computing
Platform: x86_64-w64-mingw32/x64 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library(installr)

Welcome to installr version 0.17.5

More information is available on the installr project website:
https://github.com/talgalili/installr/

Contact: <tal.galili@gmail.com>
Suggestions and bug-reports can be submitted at: https://github.com/talgalili/installr/issues

                        To suppress this message use:
                        suppressPackageStartupMessages(library(installr))

> updateR()
Installing the newest version of R,
 please wait for the installer file to be download and executed.
 Be sure to click 'next' as needed...
trying URL 'https://cran.rstudio.com/bin/windows/base/R-3.2.4revised-win.exe'
Content type 'application/x-msdos-program' length 65593847 bytes (62.6 MB)
downloaded 62.6 MB

The file was downloaded succesfully into:
 C:\Users\adminbsh\AppData\Local\Temp\RtmpqQeKyY/R-3.2.4revised-win.exe 

Running the installer now...

Installation status:  TRUE . Removing the file:
 C:\Users\adminbsh\AppData\Local\Temp\RtmpqQeKyY/R-3.2.4revised-win.exe 
 (In the future, you may keep the file by setting keep_install_file=TRUE) 

We can not seem to find the location if the new R you have installed.
The rest of the updating process is aborted, please take care to copy
your packages to the new R installation.
[1] TRUE
`
Here is the directory showing that R seems present in the right directory (account and group names munged lightly):

`  c:/R:
  total used in directory 40 available 83440560
  drwxrwxrwx  1 myname          Domain Users     4096 03-28 14:04 .
  drwxrwxrwx  1 TrustedInstallr TrustedInstallr  8192 03-22 02:28 ..
  drwxrwxrwx  1 Admin Account   Domain Users     4096 2014-07-11  R-3.1.1
  drwxrwxrwx  1 Admin Account   Domain Users     4096 2014-11-18  R-3.1.2
  drwxrwxrwx  1 Admin Account   Domain Users     4096 2015-03-31  R-3.1.3
  drwxrwxrwx  1 Admin Account   Domain Users     4096 2015-06-23  R-3.2.1
  drwxrwxrwx  1 Admin Account   Domain Users     4096 2015-08-25  R-3.2.2
  drwxrwxrwx  1 Admin Account   Domain Users     4096 12-22 08:42 R-3.2.3
  drwxrwxrwx  1 myname          Domain Users     4096 03-28 14:04 R-3.2.4revised

  c:/R/R-3.2.4revised:
  total used in directory 3136 available 83206616
  drwxrwxrwx  1 myname        Domain Users    4096 03-28 14:37 .
  drwxrwxrwx  1 myname        Domain Users    4096 03-28 14:04 ..
  -rw-rw-rw-  1 Admin Account Domain Users   28207 03-17 19:59 CHANGES
  -rw-rw-rw-  1 Admin Account Domain Users   18351 2010-03-17  COPYING
  -rw-rw-rw-  1 Admin Account Domain Users  250546 03-17 20:10 MD5
  -rw-rw-rw-  1 Admin Account Domain Users    4228 2014-07-07  README
  -rw-rw-rw-  1 Admin Account Domain Users    8952 03-17 20:09 README.R-3.2.4revised
  drwxrwxrwx  1 myname        Domain Users    4096 03-28 14:37 Tcl
  drwxrwxrwx  1 myname        Domain Users       0 03-28 14:36 bin
  drwxrwxrwx  1 myname        Domain Users    4096 03-28 14:36 doc
  drwxrwxrwx  1 myname        Domain Users    4096 03-28 14:36 etc
  drwxrwxrwx  1 myname        Domain Users    4096 03-28 14:36 include
  drwxrwxrwx  1 myname        Domain Users    4096 03-28 14:04 library
  drwxrwxrwx  1 myname        Domain Users       0 03-28 14:04 modules
  drwxrwxrwx  1 myname        Domain Users    4096 03-28 14:04 share
  drwxrwxrwx  1 myname        Domain Users       0 03-28 14:04 src
  drwxrwxrwx  1 myname        Domain Users   32768 03-28 14:37 tests
  -rw-rw-rw-  1 myname        Domain Users 1648781 03-28 14:37 unins000.dat
  -rwxrwxrwx  1 Admin Account Domain Users 1186161 03-28 14:32 unins000.exe

`
installr did make it far enough to inform RStudio about 3.2.4revised, but there are obviously no packages present. 

Any suggestions for what I do now?  What more information do you need?

Thanks,

Bill

**Comment by bharris-fs** [comment]
I tried

`> copy.packages.between.libraries(ask = T)
============= From: Choose an R version/folder from which to copy the package library ====================
  R_version       Folder
1     3.2.3 C:/R/R-3.2.3
2     3.2.2 C:/R/R-3.2.2
3     3.2.1 C:/R/R-3.2.1
4     3.1.3 C:/R/R-3.1.3
5     3.1.2 C:/R/R-3.1.2
6     3.1.1 C:/R/R-3.1.1
FROM: Write the row number of the R version/folder FROM which to copy (or move) the package library (and then press enter):
1

Thank you
============= To: Choose an R version/folder from INTO which the packages from your old R installations will be copied TO ====================
  R_version       Folder
1     3.2.3 C:/R/R-3.2.3
2     3.2.2 C:/R/R-3.2.2
3     3.2.1 C:/R/R-3.2.1
4     3.1.3 C:/R/R-3.1.3
5     3.1.2 C:/R/R-3.1.2
6     3.1.1 C:/R/R-3.1.1
TO: Write the row number of the R version/folder INTO which to copy (or move) the package library (and then press enter):`

and indeed installr doesn't seem to recognize the new 3.2.4revised version. I renamed it to 3.2.4, in case the 'revised' was problematic, but that didn't help.

I then noticed that I, not my admin account, was the owner of 3.2.4revised.  I uninstalled R-3.2.4revised and reinstalled it as admin in RGui, but I still got the same complaint from installr, despite the account now being owned by the same admin account as the other R versions.

Bill

**Comment by hdvinod** [comment]
I have exactly the same problem.

**Comment by dandermotj** [comment]
In the exact same boat here!

**Comment by talgalili** [comment]
Hi Guys,
Thank you for the bug report. This "revision" text in the name of R has caused many issues.
I will look into solving this (while it shouldn't be a problem for future releases of R, unless they use this "revision" thing again).

Tal

**Comment by talgalili** [comment]
Hi all,
This issue should be now fixed (in installr 0.17.6)
You can install it using:

```r
if (!require('devtools')) install.packages('devtools'); # make sure you have Rtools installed first! if not, then run:
#install.packages('installr')
#install.Rtools()
devtools::install_github('talgalili/installr')
```

Please let me know if this solves your issue.

Best,
Tal

**Comment by hdvinod** [comment]
your earlier workaround using the temp directory worked for me
thanks

On Sat, Apr 9, 2016 at 3:13 PM, Tal Galili <notifications@github.com> wrote:

> Hi all,
> This issue should be now fixed (in installr 0.17.6)
> You can install it using:
>
> if (!require('devtools')) install.packages('devtools'); # make sure you have Rtools installed first! if not, then run:#install.packages('installr')#install.Rtools()devtools::install_github('talgalili/installr')
>
> Please let me know if this solves your issue.
>
> Best,
> Tal
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_talgalili_installr_issues_53-23issuecomment-2D207836188&d=CwMCaQ&c=aqMfXOEvEJQh2iQMCb7Wy8l0sPnURkcqADc2guUW8IM&r=jOon43tKLVRpvfeQu95XS9U8pSo3ZLUjmqbU_jNBdQE&m=gAEMYdmWEL3Yr3vsNCig8yLTMlAfYF62GlS3yhtn_8o&s=4ap_PUSns7QOvlsopW4-R6pbptOzJ3CrR52QgC7E6-4&e=>
>



-- 

Hrishikesh (Rick) D. Vinod
Professor of Economics, Fordham University
E-Mail: Vinod@fordham.edu Tel 718-817-4065,
Secretary 718-817-4048, Fax 718-817-3518
Web page: http://www.fordham.edu/economics/vinod
ResearchGate says my papers have been cited 1894 times in various
publications.

**Comment by bharris-fs** [comment]
Tal Galili <notifications@github.com> writes:

> Hi all,
> This issue should be now fixed (in installr 0.17.6)

Thanks, Tal.  I run Linux at home, and my Windows work box is updated
already.  If I find someone else who has yet to update R on their
Windows box, I'll ask them to try installr.

Bill
-- 
Bill Harris                  
Facilitated Systems
http://makingsense.facilitatedsystems.com/
